src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -201,7 +233,6 @@ private void executeWorkflow(List<ProcessNode> workflowSequence, ActionListener<
             // Attempt to join each workflow step future, may throw a CompletionException if any step completes exceptionally
             workflowFutureList.forEach(CompletableFuture::join);
 
-            // TODO : Create State Index request with provisioning state, start time, end time, etc, pending implementation. String for now
             workflowListener.onResponse("READY");
```

Comment:
The `workflowListener` is somewhat of a placeholder that accepts a string on response. I see that the string isnt used when updating the  state index, perhaps lets just forgo the action listener all-together

Reply:
What do you suggest is then done on exception in executeWorkflow, those are still in use in catch block below.

Reply:
Removing this actually breaks the code right now, meaning we don't change state to complete or failed

Reply:
Well since we're not using the onResponse value of `READY`, we can just invoke `updateFlowFrameworkSystemIndexDoc` after `workflowFutureList.forEach(CompletableFuture::join);` and also within the catch block. 

Why use an action listener if the given value is not used onResponse?

Reply:
removed it